api/v0/ingest/schema/types_test.go

```diff
@@ -0,0 +1,156 @@
+package schema_test
+
+import (
+	"math/rand"
+	"strings"
+	"testing"
+
+	stischema "github.com/filecoin-project/storetheindex/api/v0/ingest/schema"
+	"github.com/filecoin-project/storetheindex/test/util"
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-ipld-prime"
+	_ "github.com/ipld/go-ipld-prime/codec/dagjson"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipld/go-ipld-prime/node/bindnode"
+	"github.com/ipld/go-ipld-prime/storage/memstore"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_OldSchemaIsCompatibleWithNewStructs(t *testing.T) {
+	oldSchema, err := ipld.LoadSchemaBytes([]byte(`
+type EntryChunk struct {
+	Entries List_Bytes
+	Next optional Link_EntryChunk
+}
+
+type Link_EntryChunk &EntryChunk
+type List_String [String]
+type List_Bytes [Bytes]
+
+type Advertisement struct {
+	PreviousID optional Link_Advertisement
+	Provider String
+	Addresses List_String
+	Signature Bytes
+	Entries Link
+	ContextID Bytes
+	Metadata Bytes
+	IsRm Bool
+}
+
+type Link_Advertisement &Advertisement
+`))
+	require.NoError(t, err)
+
+	rng := rand.New(rand.NewSource(1413))
+	ad := generateAdvertisement(rng)
+	oldAdType := oldSchema.TypeByName("Advertisement")
+	nodeFromOldAdType := bindnode.Wrap(&ad, oldAdType).Representation()
+
+	var newAd stischema.Advertisement
+	err = newAd.FromNode(nodeFromOldAdType)
```

Comment:
following my previous "why does FromNode copy the fields" comment - an alternative would be easier to use as well, e.g.:

    newAd, err := UnwrapAdvertisement(nodeFromOldAdType)

Reply:
Do you think keeping `LoadFrom` and `StoreTo` functions as receivers of schema types makes sense? That's also a YAGNI here, and the only reason I wrote it this way is the TODO I left about potentially proposing the interfaces to be part of IPLD.